mungegithub/mungers/submit-queue.go

```diff
@@ -681,23 +681,25 @@ func (sq *SubmitQueue) getRunningPRNumber() int {
 func (sq *SubmitQueue) monitorProw() {
 	nonBlockingJobNames := make(map[string]bool)
 	requireRetestJobNames := make(map[string]bool)
-	sq.opts.Lock()
-	for _, jobName := range sq.NonBlockingJobNames {
-		nonBlockingJobNames[jobName] = true
-	}
-	for _, jobName := range mungeopts.RequiredContexts.Retest {
-		requireRetestJobNames[jobName] = true
-	}
-	sq.opts.Unlock()
 
-	url := sq.ProwURL + "/data.js"
 	for {
+		sq.opts.Lock()
+		for _, jobName := range sq.NonBlockingJobNames {
+			nonBlockingJobNames[jobName] = true
+		}
+		for _, jobName := range mungeopts.RequiredContexts.Retest {
+			requireRetestJobNames[jobName] = true
+		}
+		url := sq.ProwURL + "/data.js"
+		sq.opts.Unlock()
+
 		currentPR := sq.getRunningPRNumber()
```

Comment:
I probably should have just locked / unlocked SQ here for running and last PR Num instead of having this method.